spec/bank_account_service_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'spec_helper'
+require_relative '../services/bank_account_service'
+require_relative '../repositories/bank_account_repository'
+require_relative '../models/bank_account'
+
+RSpec.describe BankAccountService do
+  let(:repo)    { BankAccountRepository.new }
+  let(:service) { BankAccountService.new(repo) }
+
+  before(:each) do
+    BankAccount.dataset.delete
+  end
+
+  describe '#create_account' do
+    it 'creates a new account and returns the BankAccount object' do
+      acct = service.create_account(
+        name:    'Alice',
+        job:     'Engineer',
+        email:   'alice@example.com',
+        address: '123 Main St'
+      )
+      expect(acct).to be_a(BankAccount)
+      expect(acct.name).to eq('Alice')
+      expect(acct.balance).to eq(0.0)
+    end
+
+    it 'raises a uniqueness error when email already exists' do
+      service.create_account(name: 'A', job: 'J', email: 'dup@x.com', address: 'A')
```

Comment:
I would place anything that's setting up the spec in a before hook 
https://rspec.info/features/3-12/rspec-core/hooks/before-and-after-hooks/